Mathlib/Order/Cover.lean

```diff
@@ -421,6 +421,19 @@ lemma LT.lt.exists_disjoint_Iio_Ioi (h : a < b) :
   · rcases h.exists_lt_lt h' with ⟨c, ha, hb⟩
     exact ⟨c, ha, c, hb, fun _ h₁ _ => lt_trans h₁⟩
 
+theorem covBy_iff_lt_iff_le {x y : α} : x ⋖ y ↔ ∀ z, z < y ↔ z ≤ x where
+  mp := fun hx _z ↦ ⟨hx.le_of_lt, fun hz ↦ hz.trans_lt hx.lt⟩
+  mpr := fun H ↦ ⟨(H _).2 le_rfl, fun _z hx hz ↦ ((H _).1 hz).not_lt hx⟩
+
+theorem covBy_iff_le_iff_lt {x y : α} : x ⋖ y ↔ ∀ z, z ≤ x ↔ z < y := by
+  simp_rw [covBy_iff_lt_iff_le, iff_comm]
```

Comment:
Not sure we need both this one and the one above

Reply:
The way I want to use this lemma is to rewrite the expression on the right. But I think it's perfectly reasonable to expect it to be the other way around.